internal/version/version.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package version includes constants specifying the semantic version of the
+// core Service Weaver module.
+package version
```

Comment:
Notes on what we discussed offline. We have two different versions:

1. SW module version, which includes *all* of the public API changes (not just the pipe) and is updated according to the Go module versioning strategy (e.g., on every public backward-compatible API change, bump up the minor version, on public backward-incompatible API change, bump up the major version).
2. Pipe version, which is updated only when the pipe protocol changes.

Deployers only care about pipe versions. We decided therefore to change the version in this file to represent the pipe version, with the following properties:

* Pipe version is based on the SW module version that introduced the version change.
* Pipe version follows the same major/minor/patch properties as the SW module version.
* Changes to the SW module's major version trigger changes to the pipe version, and vice-versa.
* Changes to the SW module's minor version don't trigger changes to the pipe version.
* Changes to the pipe minor version trigger changes to the SW module's minor version.

I think all of the above means that, when pipe versions are incompatible, the deployer can print out the weavelet's pipe version, which matches some version of SW module, and ask the user to install a deployer that supports that particular SW module version. Deployers can keep a list of supported SW module versions. That way, the user doesn't have to deal or know what a "pipe version" really is.

Reply:
Thanks for the offline discussion Srdjan. It helped clarify and solidify a lot of my thoughts. I updated all the comments in this PR to reflect the new model. I also moved the version into the `runtime` package because I realized that other deployers will likely need to use it as well. I'll follow up with a doc describing everything we talked about in more detail, and I'll be sure to include your comment.